### ISSUE TITLE
Detect upstream git branch based on git config

### ISSUE THREAD

**Issue by ThisIsMissEm** [azz/pretty-quick#114]
Currently, this tool hardcodes the usage of `master` as the upstream branch, which can often not be the case.

https://github.com/azz/pretty-quick/blob/master/src/scms/git.js#L31-L37

```
const revision = staged
      ? 'HEAD'
      : runGit(directory, [
          'merge-base',
          'HEAD',
          branch || 'master',
        ]).stdout.trim();
```

You can potentially provide better user experience by first checking if the user has configured a different default branch for their repos: `git config --get init.defaultbranch`, if that returns a value, then use that value, otherwise, use `master` (or if you wanna be progressive, `main`, but that'd be a breaking change — n.b., GitHub has published intent to change the default branch to always be `main` not `master`: https://github.com/github/renaming )

**Comment by JounQin** [comment]
Yes, it can be improved, currently you can use `--branch` option to override.
